src/scripts/librarymenu.js

```diff
@@ -155,297 +302,669 @@ define(["dom", "layoutManager", "inputManager", "connectionManager", "events", "
-        })
+        if (!user) {
+            showBySelector(".libraryMenuDownloads", false);
+            showBySelector(".lnkSyncToOtherDevices", false);
+            return void showBySelector(".userMenuOptions", false);
+        }
+
+        if (user.Policy.EnableContentDownloading) {
+            showBySelector(".lnkSyncToOtherDevices", true);
+        } else {
+            showBySelector(".lnkSyncToOtherDevices", false);
+        }
+
+        if (user.Policy.EnableContentDownloading && appHost.supports("sync")) {
+            showBySelector(".libraryMenuDownloads", true);
+        } else {
+            showBySelector(".libraryMenuDownloads", false);
+        }
+
+        var userId = Dashboard.getCurrentUserId();
+        var apiClient = getCurrentApiClient();
+        var libraryMenuOptions = document.querySelector(".libraryMenuOptions");
+
+        if (libraryMenuOptions) {
+            getUserViews(apiClient, userId).then(function (result) {
+                var items = result;
+                var html = "";
+                html += '<h3 class="sidebarHeader">';
+                html += globalize.translate("HeaderMedia");
+                html += "</h3>";
+                html += items.map(function (i) {
+                    var icon = "folder";
+                    var itemId = i.Id;
+
+                    if ("channels" == i.CollectionType) {
+                        itemId = "channels";
+                    }
+                    else if ("livetv" == i.CollectionType) {
+                        itemId = "livetv";
+                    }
+
+                    else if ("photos" == i.CollectionType) {
+                        icon = "photo_library";
+                    }
+                    else if ("music" == i.CollectionType || "musicvideos" == i.CollectionType) {
+                        icon = "library_music";
+                    }
+                    else if ("books" == i.CollectionType) {
+                        icon = "library_books";
+                    }
+                    else if ("playlists" == i.CollectionType) {
+                        icon = "view_list";
+                    }
+                    else if ("games" == i.CollectionType) {
+                        icon = "games";
+                    }
+                    else if ("movies" == i.CollectionType) {
+                        icon = "video_library";
+                    }
+                    else if ("channels" == i.CollectionType || "Channel" == i.Type) {
+                        icon = "videocam";
+                    }
+                    else if ("tvshows" == i.CollectionType) {
+                        icon = "tv";
+                    }
+                    else if ("livetv" == i.CollectionType) {
+                        icon = "live_tv";
+                    }
+                    icon = i.icon || icon;
+
+                    if (i.onclick) {
+                        i.onclick;
```

Comment:
what is this?..

Reply:
No idea. Used to be a huuuge amount of if/elses nested inside each other.

Just made a lazy equivalent to it. If you have a suggested change, I can merge it in.